proximal/tests/test_algs.py

```diff
@@ -403,6 +403,28 @@ def test_lin_admm_two_prox_fn(self):
                            eps_rel=1e-5)
         self.assertItemsAlmostEqual(x.value, cvx_X.value, eps=2e-2)
 
+    def test_lin_admm_two_prox_fn_shape_matching(self):
+        # With nested linear operators.
+        kernel_mat = np.array([[2, 1, 3], [3, 2, 1], [1, 3, 2]])
+
+        N = 3
+        factor = 1
+        b = np.ones((N, N))
+        x = px.Variable((N * factor, N * factor))
+        prox_fns = [
+            px.norm1(x),
+            px.sum_squares(px.subsample(px.conv(kernel_mat, x),
+                                        (factor, factor)),
+                           b=b),
+        ]
+        psi_fns, omega_fns = ladmm.partition(prox_fns)
+        sltn = ladmm.solve(psi_fns,
```

Comment:
Shouldn't you verify the solution?

Reply:
Hi @SteveDiamond I feel conflicted whether to verify the solution by `assert norm2(cvx_solution, proxomal_solution)`. The original goal of the test was to expose the shape mismatch exception. I checked and it seemed the best way to assert for no exception is do do something like this: https://stackoverflow.com/a/20275035

I am not sure if numerical precision is the goal here... One specific application involving the model `subsample(conv(h, x), K)` is the TIRF / STED molecule localization microscopy. The algorithm, sparsifying deconvolution algorithm, emphasize the location calls (aka index of non-zero values) are than the recovered intensity values in the solutions.

How do we verify the solution with that of `cvxpy`, but to avoid a false negative of the test in a long run? If you can show me a pseudo-code, I am pretty sure I can update the test code accordingly. Thank you

Reply:
Ok, that's fair, you're specifically testing that the shape error doesn't come up. I guess testing the solution isn't necessary.